packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -0,0 +1,93 @@
+import { Aliases, AuthInfo } from '@salesforce/core';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { ForceConfigGet } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+
+export async function setupWorkspaceOrgType() {
+  const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  let isScratchOrg = false;
+  if (defaultUsernameIsSet) {
+    const username = await getUsername(defaultUsernameOrAlias!);
+    try {
+      isScratchOrg = await isAScratchOrg(username);
+    } catch (e) {
+      if (e.name === 'NamedOrgNotFound') {
+        // TODO: what should we do in this case? Expose all source commands by default?
+        setDefaultOrgIsScratchOrg(true);
+        setDefaultOrgIsNonScratchOrg(true);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+  setDefaultOrgIsScratchOrg(defaultUsernameIsSet && isScratchOrg);
+  setDefaultOrgIsNonScratchOrg(defaultUsernameIsSet && !isScratchOrg);
+}
+
+async function isAScratchOrg(username: string): Promise<boolean> {
+  const authInfo = await AuthInfo.create(username);
+  const authInfoFields = authInfo.getFields();
+  return Promise.resolve(typeof authInfoFields.devHubUsername !== 'undefined');
+}
+
+/**
+ * Returns the non-aliased username
+ * @param usernameOrAlias
+ */
+async function getUsername(usernameOrAlias: string): Promise<string> {
+  const username = await Aliases.fetch(usernameOrAlias);
+  if (username) {
+    return Promise.resolve(username);
+  }
+  return Promise.resolve(usernameOrAlias);
+}
+
+function setDefaultOrgIsScratchOrg(val: boolean) {
```

Comment:
Might it be better to track based on source tracking availability vs whether it's a scratch org?
That might make this  more future proof to when we actually add source tracking to other non-scratch orgs.
So, the action item here might just be to name things like _setDefaultUsernameHasChangeTracking_.
I think it's called `defaultusername` right in .sfdx/sfdx-config.json?

Reply:
Yep, that makes sense.